tutorials/parameter_sweep_demo.ipynb

```diff
@@ -0,0 +1,766 @@
+    "*Generate samples from a distribution or a Euclidean space and solve flowsheets with those input values.*"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fe985f68",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Primary requirements for running a parameter sweep\n",
+    "\n",
+    "* Function to construct the Pyomo model of the flowsheet\n",
+    "* Function to create the sweep parameters\n",
+    "* Initialization and optimization functions (reinitialization function deprecated)\n",
+    "* Parallel computing information\n",
+    "* Output file information"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2fd561d9",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Test Flowsheet\n",
+    "\n",
+    "* We use a simple **RO system with an energy recovery device** to demonstrate the capabilities of the parameter sweep tool. \n",
+    "* It comprises of an inlet high pressure pump connected to a steady state zero-dimensional RO process model. A turbine-type isothermal energy recovery device model is connected downstream of the RO model.\n",
+    "* The feed water is an NaCl brine solution and is modeled using an property package within WaterTAP. \n",
+    "* Outputs that can be measured from this flow sheet include LCOW, product flow rate and concentration, volumetric recovery, water recovery, and specific energy consumption.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3551ef32",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "subslide"
+    }
+   },
+   "source": [
+    "<div>\n",
+    "    <img src=\"assets_parameter_sweep_demo/RO_ERD_flowsheet.png\" alt=\"RO with energy recovery device\" width=\"60%\" height=\"auto\" align=\"center\"/>\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "53ce93e7",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Make the necessary imports\n",
+    "from pprint import pprint\n",
+    "from IPython import get_ipython\n",
+    "from idaes.core.solvers import get_solver\n",
+    "from watertap.examples.flowsheets.RO_with_energy_recovery.RO_with_energy_recovery import (\n",
+    "    optimize,\n",
+    ")\n",
+    "from watertap.examples.flowsheets.RO_with_energy_recovery.monte_carlo_sampling_RO_ERD import (\n",
+    "    build_model,\n",
+    "    build_outputs,\n",
+    ")\n",
+    "from watertap.tools.parameter_sweep import (\n",
+    "    LinearSample,\n",
+    "    ParameterSweep,\n",
+    ")\n",
+    "from assets_parameter_sweep_demo.parameter_sweep_demo_script import build_sweep_params"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a9de9f80",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "subslide"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "??build_model"
```

Comment:
Can you check on whats going on with these

Reply:
That is just an iPython command to show the function definition